plugin/dapp/x2Ethereum/executor/x2ethereumaction.go

```diff
@@ -0,0 +1,660 @@
+	if oracleKeeper == nil {
+		return nil
+	}
+
+	elog.Info("newAction", "newAction", "done")
+	return &action{a.GetAPI(), a.GetCoinsAccount(), a.GetStateDB(), hash, fromaddr,
+		a.GetBlockTime(), a.GetHeight(), index, address.ExecAddress(string(tx.Execer)), ethbridge.NewKeeper(*oracleKeeper, a.GetStateDB())}
+}
+
+// ethereum ---> chain33
+// lock
+func (a *action) procMsgEth2Chain33(ethBridgeClaim *types2.Eth2Chain33) (*chain33types.Receipt, error) {
+	receipt := new(chain33types.Receipt)
+	ethBridgeClaim.LocalCoinSymbol = strings.ToLower(ethBridgeClaim.LocalCoinSymbol)
+
+	consensusNeededBytes, err := a.db.Get(types2.CalConsensusThresholdPrefix())
+	if err != nil {
+		if err == chain33types.ErrNotFound {
+			setConsensusThreshold := &types2.ReceiptQueryConsensusThreshold{ConsensusThreshold: types2.DefaultConsensusNeeded}
+			msgSetConsensusThresholdBytes, err := proto.Marshal(setConsensusThreshold)
+			if err != nil {
+				return nil, chain33types.ErrMarshal
+			}
+			receipt.KV = append(receipt.KV, &chain33types.KeyValue{
+				Key:   types2.CalConsensusThresholdPrefix(),
+				Value: msgSetConsensusThresholdBytes,
+			})
+			consensusThreshold := &types2.ReceiptSetConsensusThreshold{
+				PreConsensusThreshold: int64(0),
+				NowConsensusThreshold: types2.DefaultConsensusNeeded,
+				XTxHash:               a.txhash,
+				XHeight:               uint64(a.height),
+			}
+			receipt.Logs = append(receipt.Logs, &chain33types.ReceiptLog{Ty: types2.TySetConsensusThresholdLog, Log: chain33types.Encode(consensusThreshold)})
+		} else {
+			return nil, err
+		}
+	} else {
+		var mc types2.ReceiptQueryConsensusThreshold
+		_ = proto.Unmarshal(consensusNeededBytes, &mc)
+		_, _, err = a.keeper.ProcessSetConsensusNeeded(mc.ConsensusThreshold)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	status, err := a.keeper.ProcessClaim(*ethBridgeClaim)
+	if err != nil {
+		return nil, err
+	}
+
+	ID := strconv.Itoa(int(ethBridgeClaim.EthereumChainID)) + strconv.Itoa(int(ethBridgeClaim.Nonce)) + ethBridgeClaim.EthereumSender + ethBridgeClaim.TokenContractAddress + "lock"
+
+	//记录ethProphecy
+	bz, err := a.db.Get(types2.CalProphecyPrefix(ID))
+	if err != nil {
+		return nil, types2.ErrProphecyGet
+	}
+
+	var dbProphecy types2.ReceiptEthProphecy
+	err = proto.Unmarshal(bz, &dbProphecy)
+	if err != nil {
+		return nil, chain33types.ErrUnmarshal
+	}
+
+	receipt.KV = append(receipt.KV, &chain33types.KeyValue{
+		Key:   types2.CalProphecyPrefix(ID),
+		Value: bz,
+	})
+	receipt.Logs = append(receipt.Logs, &chain33types.ReceiptLog{Ty: types2.TyProphecyLog, Log: chain33types.Encode(&dbProphecy)})
+
+	if status.Text == types2.EthBridgeStatus_SuccessStatusText {
+		// mavl-x2ethereum-eth
+		// 这里为了区分相同tokensymbol不同tokenAddress做了级联处理
+		accDB, err := account.NewAccountDB(a.api.GetConfig(), ethBridgeClaim.LocalCoinExec, strings.ToLower(ethBridgeClaim.LocalCoinSymbol+ethBridgeClaim.TokenContractAddress), a.db)
```

Comment:
localCoinExec采用常量，不要使用参数
```suggestion
		accDB, err := account.NewAccountDB(a.api.GetConfig(), ethBridgeClaim.LocalCoinExec, strings.ToLower(ethBridgeClaim.LocalCoinSymbol+ethBridgeClaim.TokenContractAddress), a.db)
```